app/plugin.go

```diff
@@ -257,9 +253,7 @@ func (a *App) SyncPlugins() *model.AppError {
 }
 
 func (a *App) ShutDownPlugins() {
-	a.Srv.PluginsLock.Lock()
-	pluginsEnvironment := a.Srv.PluginsEnvironment
-	defer a.Srv.PluginsLock.Unlock()
+	pluginsEnvironment := a.GetPluginsEnvironment()
```

Comment:
Won't releasing the lock when sending `OnDeactivate` event be susceptible to the same issue of another plugin spinning up while this one is shutting down? Albeit the window is a lot tinier.

Reply:
@lieut-data would someone have to invoke to start setting up another plugins env or can that start up implicitly on some sort of config change? Wondering if it's worth the hassle to pull out `OnDeactivate`, as I'll need to update `RunMultiPluginHook(hookRunnerFunc func(hooks Hooks) bool, hookId int)` to `RunMultiPluginHook(hookRunnerFunc func(hooks Hooks, *pluginId string*) bool, hookId int)` as I'll need to expose pluginId in case of an error when calling `OnDeactivate`.